scripts/polarion_sync/push_gate.py

```
@@ -0,0 +1,349 @@
+    # Diff metadata lines
+    re.compile(r"^(\+\+\+|---|\@\@|diff )"),
+    # Import lines (pytest import may be added)
+    re.compile(r"^\+\s*import\s"),
+    re.compile(r"^\+\s*from\s"),
+)
+
+
+def _run_precommit(repo_root: Path) -> bool:
+    """Run pre-commit and return True if it passes."""
+    LOGGER.info("Running pre-commit...")
+    result = subprocess.run(
```

Ruff (0.15.15)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)